pkg/node/node_address_linux.go

```diff
@@ -17,23 +17,27 @@
 package node
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 	"net"
+	"sort"
 
 	"golang.org/x/sys/unix"
 
 	"github.com/vishvananda/netlink"
 )
 
-func firstGlobalV4Addr(intf string) (net.IP, error) {
+// firstGlobalV4Addr  returns the first IP address for the given interface, if
+// the wishIP is send it'll return the wishIP if is in the list of address.
+func firstGlobalV4Addr(intf string, wishIP net.IP) (net.IP, error) {
```

Comment:
It's in the name but the comment might be a little more clear with
```
firstGlobalV4Addr returns the first IPv4 global IP of an interface, where the IPs are sorted in ascending order. when intf is defined only IPs belonging to that interface are considered. If wishIP is present in the IP list it is returned irrespective of the sort order. Passing intf and wishIP will only return wishIP if it is in the IPs that belong to intf. In all cases, if intf is not found all interfaces are considered.
```
(something similar will apply to the v6 one below)